daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/ComplexTypes.scala

```diff
@@ -54,6 +54,16 @@ sealed abstract class ComplexTypeBase(xmlArg: Node, parentArg: SchemaComponent)
     s
   }
 
+  modelGroup match {
+    case s: SequenceGroupTermBase =>
+      schemaDefinitionWhen(
```

Comment:
Does it make sense to do `s.schemaDefinitionWhen` so that the context of the SDE is the xs:sequence line that contains hiddenGroupRef instead of the xs:complexType line? I guess in reality they will always be right next to each other, but since the problem is really the xs:sequence that maybe gives a slightly more accurate diagnostic?